src/plugins/saved_objects_management/public/management_section/objects_table/components/duplicate_modal.tsx

```diff
@@ -0,0 +1,467 @@
+    const { savedObjectTypeInfoMap } = this.state;
+    const savedObjectTypeInfo = savedObjectTypeInfoMap.get(savedObjectType);
+    if (savedObjectTypeInfo) {
+      const [count, checked] = savedObjectTypeInfo;
+      savedObjectTypeInfoMap.set(savedObjectType, [count, !checked]);
+      this.setState({ savedObjectTypeInfoMap });
+    }
+  };
+
+  renderDuplicateObjectCategory = (
+    savedObjectType: string,
+    savedObjectTypeCount: number,
+    savedObjectTypeChecked: boolean
+  ) => {
+    return (
+      <EuiCheckbox
+        id={'includeSavedObjectType.' + savedObjectType}
+        key={savedObjectType}
+        label={
+          <FormattedMessage
+            id={
+              'savedObjectsManagement.objectsTable.duplicateModal.savedObjectType.' +
+              savedObjectType
+            }
+            defaultMessage={
+              capitalizeFirstLetter(savedObjectType) + ` (${savedObjectTypeCount.toString()})`
+            }
+          />
+        }
+        checked={savedObjectTypeChecked}
+        onChange={() => this.changeIncludeSavedObjectType(savedObjectType)}
+      />
+    );
+  };
+
+  renderDuplicateObjectCategories = () => {
+    const { savedObjectTypeInfoMap } = this.state;
+    const checkboxList: JSX.Element[] = [];
+    savedObjectTypeInfoMap.forEach(
+      ([savedObjectTypeCount, savedObjectTypeChecked], savedObjectType) =>
+        checkboxList.push(
+          this.renderDuplicateObjectCategory(
+            savedObjectType,
+            savedObjectTypeCount,
+            savedObjectTypeChecked
+          )
+        )
+    );
+    return checkboxList;
+  };
+
+  isSavedObjectTypeIncluded = (savedObjectType: string) => {
+    const { savedObjectTypeInfoMap } = this.state;
+    const savedObjectTypeInfo = savedObjectTypeInfoMap.get(savedObjectType);
+    return savedObjectTypeInfo && savedObjectTypeInfo[1];
+  };
+
+  render() {
+    const {
+      workspaceOptions,
+      targetWorkspaceOption,
+      isIncludeReferencesDeepChecked,
+      allSelectedObjects,
+    } = this.state;
+    const { duplicateMode, onClose } = this.props;
+    const targetWorkspaceId = targetWorkspaceOption?.at(0)?.key;
+    let selectedObjects = allSelectedObjects;
+    if (duplicateMode === DuplicateMode.All) {
+      selectedObjects = selectedObjects.filter((item) => this.isSavedObjectTypeIncluded(item.type));
+    }
+    const includedSelectedObjects = selectedObjects.filter((item) =>
+      !!targetWorkspaceId && !!item.workspaces ? !item.workspaces.includes(targetWorkspaceId) : true
+    );
+
+    const ignoredSelectedObjectsLength = selectedObjects.length - includedSelectedObjects.length;
+
+    let confirmDuplicateButtonEnabled = false;
+    if (!!targetWorkspaceId && includedSelectedObjects.length > 0) {
+      confirmDuplicateButtonEnabled = true;
+    }
+
+    const warningMessageForOnlyOneSavedObject = (
+      <p>
+        <b style={{ color: '#000' }}>1</b> saved object will <b style={{ color: '#000' }}>not</b> be
```

Comment:
https://oui.opensearch.org/1.5/#/display/text#coloring-text . Do not hard code color as we need to take dark mode into consideration.

Reply:
Yes, I will modify the code.